src/bcd/bcd/IDenoiser.h

```diff
@@ -13,15 +13,29 @@
 #ifndef I_DENOISER_H
 #define I_DENOISER_H
 
-#include <functional>
-#include <memory>
 
 namespace bcd
 {
 
 template<class T>
 class DeepImage;
 
+class IProgressReporter
```

Comment:
Should probably be moved to appleseed.foundation no?

In which case `isAborted()` should be renamed to `is_aborted()`.

Reply:
I didn't want to introduce a dependency between BCD and foundation.
This is BCD's progress reporter and there is an adaptor to use our abort switch and possibly report progress.

Reply:
Didn't we already introduce dependencies to appleseed in BCD?